### PR TITLE
Added basic support for API push

### PR DESCRIPTION
**Pull request by ABeltramo**
It's working for my use case (see [issue 42](https://github.com/jariz/Noti/issues/42) ) i don't know if it's the solution to also  [issue 21](https://github.com/jariz/Noti/issues/21).
For sure it's a starting point to add other useful notification to Noti.
The line `now -= 5` it's a little trick and I wan't to find a better code when I have the time.
Hope to help others! Good work btw..

### CODE REVIEW

**Comment by jariz** [comment]
Hey, very glad you're contributing, but if we'd add this, I'd want to do this properly.  
Besides this being very hacky, what if the user clicks notifications?  
I myself when using pushes only use it to send links to my device and vice versa (and I think a lot of people do), so this would be a must have.  
Regarding the `-= 5` hack, I'm not even sure why it's needed. You can simply save the timestamp (w/`UserDefaults`) of the last time a push tickle happened and use that for `modified_after` in the `/pushes/` call.

**Comment by ABeltramo** [comment]
As I sad it's a starting point 😃 
What do you propose to do on click? For example I use it to get notification from a server machine not to send link. Autodetect content?
For the `-= 5` hack I have to take better look at the API. That's the first simple solution to check if my code works.

**Comment by jariz** [comment]
Pushes with a link have a optional 'url' property, so that would be really easy to use. 😄

**Comment by ABeltramo** [comment]
I made a few changes on the code. 

- Now the hack it's gone, I get only the latest post from the API. 
- When you receive a link you can press the show button to open it on the browser. 
- When you receive an image you can open it on the default browser.

**Todo (need help)**

- Can't understand how file in general works. If I open the fileurl link it gives me an error.
- Does it work with encryption? I think not but I don't know if message to yourself will be encrypted

**Comment by jariz** [comment]
Looks good :)  
I'll see if I can help you with those other points this week.

File: Noti/PushManager.swift [hunk 1]
```diff
@@ -459,6 +431,76 @@ class PushManager: NSObject, WebSocketDelegate, NSUserNotificationCenterDelegate
         
     }
     
+    // Given a push JSON object from Pushbullet create a notification
+    // Used in mirror and subtipe:push
+    internal func createNotification(push : JSON) -> NSUserNotification{
+        let notification = NSUserNotification()
+        notification.actionButtonTitle = "Show"
+        if let url = push["url"].string {                       // Special type of Noti: url type
```
Comment (jariz) [inline Noti/PushManager.swift]: wouldn't it make more sense to check on `push["type"]`? (possible values "note", "file", "link" & "mirror")

Comment (ABeltramo) [inline Noti/PushManager.swift]: Yep! Next commit I'll clean the code! 
Can I correct also all the JSON.parse() calls? It gives me 10 warning that is deprecated.

**Comment by ABeltramo** [comment]
Little update: both message send from app and from website with encryption on seems to be send in plain text. Seems we don't need to decrypt for this kind of message.
Can you confirm this?

**Comment by ABeltramo** [comment]
Everything it's completed, I think it's ready to be merged! 🏁

**Comment by cbarratt** [comment]
Was looking to see whether Noti had this functionality! Looks good, waiting for a release on this :)

**Comment by ABeltramo** [comment]
Closing this. We are refactoring in Swift 4

**Pull request closed without merging**